File: Jul15/Troy/Q1.java
class Solution {
    public int pivotIndex(int[] nums) {
            int sumRHS = sumArray(nums);
            int sumLHS = 0;
            for(int i = 0; i < nums.length; i++) {
                if (sumLHS == (sumRHS - nums[i])) {
                    return i;
                }
                sumLHS += nums[i];
                sumRHS -= nums[i];
            }
        return -1;
    }
    public int sumArray(int[] nums) {
        int result = 0;
        for (int i = 0; i < nums.length; i++) {
            result += nums[i];
        }
        return result;
    }
}

File: 2018/Jul15/Carl/Q1.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        leftSum = 0
        rightSum = sum(nums)
        for i, value in enumerate(nums):
            rightSum -= value
            if(leftSum == rightSum):
                return i
            leftSum += value
        return -1


s = Solution()
print(s.pivotIndex([-1, -1, 0, 1, 1, -1]))
